### PR TITLE
Refactor form generation into block handler

### PR DESCRIPTION
**Pull request by ajmaddaford**
### What is the context of this PR?
Moves the form generation into the block handler. As well as the block handler being the more appropriate place for the form generation, this also contributes to the aim of removing logic from the top level route file.

### How to review 
- Do you agree with the implementation?
- Should further refactoring be done?

### Checklist

* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

File: app/views/handlers/block.py [hunk 1]
```diff
@@ -114,3 +109,10 @@ def _update_section_completeness(self, location: Optional[Location] = None):
             section_id=location.section_id,
             list_item_id=location.list_item_id,
         )
+
+    def _set_started_at_metadata(self):
+        collection_metadata = self._questionnaire_store.collection_metadata
+        if not collection_metadata.get("started_at"):
+            started_at = datetime.utcnow().isoformat()
+            logger.info("Survey started", started_at=started_at)
+            collection_metadata["started_at"] = started_at
```
Comment (iwootten) [inline app/views/handlers/block.py:118]: Mutating the metadata in this way doesn't seem great. Can't started_at be set when the collection_metadata is first created?

Comment (ajmaddaford) [reply app/views/handlers/block.py:118]: This isn't metadata, this is collection metadata that was created for this purpose. It's done this way so that it is only set once they have POSTed a page. This also isn't new, it was previously called from the route.

File: app/forms/questionnaire_form.py [hunk 2]
```diff
@@ -377,21 +374,43 @@ def _get_error_id(answer_id):
     return f"{answer_id}-error"
 
 
+def _clear_detail_answer_field(form_data, question):
+    """
+    Checks the submitted answers and in the case of both checkboxes and radios,
```
Comment (LJBabbage) [inline app/forms/questionnaire_form.py:379]: I found the comment a bit hard to follow and a little verbose. Isn't it just "clear detail_answer if parent option not selected"? Also I am not sure we need the params, nothing else in that file states them and they were obvious to me

File: app/views/handlers/question.py [hunk 3]
```diff
@@ -108,9 +139,9 @@ def get_last_viewed_question_guidance_context(self):
             return {"first_location_in_section_url": first_location_in_section_url}
 
     def handle_post(self):
-        self.questionnaire_store_updater.update_answers(self.form)
-
+        self.questionnaire_store_updater.update_answers(self.form.data)
         self.questionnaire_store_updater.add_completed_location()
+        self._set_started_at_metadata()
```
[the remark this answers is not in the document]
Comment (ajmaddaford) [reply app/views/handlers/question.py:144]: Yes, if `handle_post` is called it means something has been POSTed and there were no validation errors.

Good spot on the consistency between the two handlers. There was duplication in the `Question` handlers `handle_post` method, so I have removed that and added a call to the parent block handers `handle_post` method.
